guides/21_Background_Filter_VideoFX_Processor.md

```diff
@@ -0,0 +1,375 @@
+
+**Note**  
+To complete those steps, you must first: 
+* Create a `Logger`\.
+* Choose a video device of class `MediaDeviceInfo`\.
+* Successfully join a `MeetingSession`\.
+
+Steps in the following sections explain how to complete the process\.
+
+## Starting video input<a id="start-video-input"></a>
+
+The following example shows how to use the `VideoTransformDevice` object to start video input\. 
+
+```
+// assuming that meetingSession has already been created
+await meetingSession.audioVideo.startVideoInput(videoTransformDevice);
+meetingSession.audioVideo.start();
+meetingSession.audioVideo.startLocalVideoTile();
+```
+
+## Checking for support before offering a filter<a id="support-check"></a>
+
+The Amazon Chime SDK provides an asynchronous static method that checks for supported browsers and attempts to download the required assets\. However, it does not check for device performance\. As a best practice, always ensure the users' browsers and devices can support the filters before you offer the filters\.
+
+```
+import {
+    VideoFxProcessor
+} from 'amazon-chime-sdk-js';
+
+if (!await VideoFxProcessor.isSupported(logger)) {     
+    // logger is optional for isSupported
+}
+```
+
+## Creating a VideoFxConfig object<a id="create-videofxconfig"></a>
+
+You can define configurations for `backgroundBlur` and `backgroundReplacement` in the same object\. However, you can't set `isEnabled` to `true` for both filters at the same time\. That's an invalid configuration\.
+
+The `VideoFxConfig` class does no validation of its own\. Validation occurs in the next step\.
+
+The following example shows a valid `VideoFxConfig`\.
+
+```
+const videoFxConfig: VideoFxConfig = {
+    backgroundBlur: {
+        isEnabled: false,
+        strength: 'medium'
+    },
+    backgroundReplacement: {
+        isEnabled: false,
+        backgroundImageURL: 'space.jpg',
+        defaultColor: undefined,
+    }
+}
+```
+
+## Creating a VideoFxProcessor object<a id="create-videofxprocessor"></a>
+
+When creating the VideoFxProcessor object, AWS servers download the runtime assets, or a browser cache loads the assets\. If network or CSP configurations prevent access to the assets, the `VideoFxProcessor.create` operation throws an exception\. The resulting VideoFxProcessor is configured as a no\-op processor, which won’t affect the video stream\.
+
+```
+let videoFxProcessor: VideoFxProcessor | undefined = undefined;
+try {
+  videoFxProcessor = await VideoFxProcessor.create(logger, videoFxConfig);
+} catch (error) {
+  logger.warn(error.toString());
+}
+```
+
+`VideoFxProcessor.create` also attempts to load the image from `backgroundReplacement.backgroundImageURL`\. If the image fails to load, the processor throws an exception\. The processor also throws exceptions for other reasons, such as invalid configurations, unsupported browsers, or underpowered hardware\. 
+
+## Configuring the VideoFxProcessor object<a id="configure-videofxprocessor"></a>
```

Comment:
Shouldn't this section be below or the same as creating a videoFXConfig object?

Reply:
it is in the same level as creating a `videoFxConfig` object. However you need to create a `videoFxConfig` first, then creating a `VideoFxProcessor`

Reply:
I mean you would want to explain all the config properties in the above section `Creating a VideoFxConfig`. Not sure why we have a section about creating a video Fx config without explaining any of props then explain them 2 sections later.

Reply:
Yeah that makes sense. Updated it!